lnbits/wallets/breez.py

```diff
@@ -0,0 +1,257 @@
+            breez_event_queue.put_nowait(event)
+
+    class BreezSdkWallet(Wallet):  # type: ignore[no-redef]
+        def __init__(self):
+            if not settings.breez_greenlight_seed:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: missing breez_greenlight_seed"
+                )
+            if not settings.breez_api_key:
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: missing breez_api_key"
+                )
+            if (
+                settings.breez_greenlight_device_key
+                and not settings.breez_greenlight_device_cert
+            ):
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: "
+                    "missing breez_greenlight_device_cert"
+                )
+            if (
+                settings.breez_greenlight_device_cert
+                and not settings.breez_greenlight_device_key
+            ):
+                raise ValueError(
+                    "cannot initialize BreezSdkWallet: "
+                    "missing breez_greenlight_device_key"
+                )
+
+            self.config = breez_sdk.default_config(
+                breez_sdk.EnvironmentType.PRODUCTION,
+                settings.breez_api_key,
+                breez_sdk.NodeConfig.GREENLIGHT(
+                    config=breez_sdk.GreenlightNodeConfig(
+                        partner_credentials=load_greenlight_credentials(),
+                        invite_code=settings.breez_greenlight_invite_code,
+                    )
+                ),
+            )
+
+            breez_sdk_working_dir = Path(settings.lnbits_data_folder, "breez-sdk")
+            breez_sdk_working_dir.mkdir(parents=True, exist_ok=True)
+            self.config.working_dir = breez_sdk_working_dir.absolute().as_posix()
+
+            try:
+                seed = breez_sdk.mnemonic_to_seed(settings.breez_greenlight_seed)
+                connect_request = breez_sdk.ConnectRequest(self.config, seed)
+                self.sdk_services = breez_sdk.connect(connect_request, SDKListener())
+            except Exception as exc:
+                raise ValueError(f"cannot initialize BreezSdkWallet: {exc!s}") from exc
+
+        async def cleanup(self):
+            self.sdk_services.disconnect()
+
+        async def status(self) -> StatusResponse:
+            try:
+                node_info: breez_sdk.NodeState = self.sdk_services.node_info()
+            except Exception as exc:
+                return StatusResponse(f"Failed to connect to breez, got: '{exc}...'", 0)
+
+            return StatusResponse(None, int(node_info.channels_balance_msat))
+
+        async def create_invoice(
+            self,
+            amount: int,
+            memo: Optional[str] = None,
+            description_hash: Optional[bytes] = None,
+            unhashed_description: Optional[bytes] = None,
+            **kwargs,
+        ) -> InvoiceResponse:
```

Comment:
add `try-except` and return `return InvoiceResponse(False, None, None, str(e))` on `except`